src/validate_data.py

```diff
@@ -0,0 +1,98 @@
+# clean_validate.py
+# author: Weilin Han
+# date: 2024-10-03
+
+import pandas as pd
+import pandera as pa
+
+# Function to build schema from the config file
+def build_schema_from_csv(data_config, expected_columns):
+    """Building schema for validation"""
+
+    # Ensure the data_config is a pandas dataframe
+    if not isinstance(data_config, pd.DataFrame):
+        raise TypeError("data_config must be a pandas dataframe.")
+    
+    # Ensure the data_config has following columns: column,type,max,min,category
+    required_columns = ['column', 'type', 'min', 'max','category', 'max_nullable']
+    missing_columns = [col for col in required_columns if col not in data_config.columns]
+    if missing_columns:
+        raise ValueError(f"The data_config must have following columns: 'column', 'type', 'min', 'max', 'category', 'max_nullable'.")
+
+    # Ensure the values of 'column' match the column names extracted from name file
+    if expected_columns is not None:
+        actual_columns = data_config['column'].str.strip("'").tolist()  # Clean up any extra quotation marks in 'column'
+        if actual_columns != expected_columns:
+            raise ValueError("Column names in the config file do not match the expected columns.")
+    
+
+    schema_dict = {}
+    
+    # Loop through each row in the config DataFrame
+    for _, row in data_config.iterrows():
+        column_name = row['column'].strip()  # Removing potential extra spaces
+        column_type = row['type'].strip()    # Strip any spaces
+        min_value = row['min'] if pd.notna(row['min']) else None
+        max_value = row['max'] if pd.notna(row['max']) else None
+        category_in = row['category'] if pd.notna(row['category']) else None
+        max_nullable = row['max_nullable'] if pd.notna(row['max_nullable']) else None
+        
+        # Define the correct Pandera data type
+        if column_type == 'int':
+            dtype = pa.Int
+        elif column_type == 'float':
+            dtype = pa.Float
+        elif column_type == 'str':
+            dtype = pa.String
+        else:
+            raise ValueError(f"Unsupported column type: {column_type}")
```

Comment:
None of this is needed. Pandera's schemas work with `int`, or `float`, or `str`. See an example here: https://github.com/ttimbers/breast-cancer-predictor/blob/206d1c2ba56583e87dbc359538c007698df4772c/src/validate_data.py#L45